cfme/infrastructure/cluster.py

```diff
@@ -4,48 +4,140 @@
+    totals_for_hosts = SummaryTable(title='Totals for Hosts')
+    totals_for_vms = SummaryTable(title='Totals for VMs')
+    configuration = SummaryTable(title='Configuration')
+    smart_management = SummaryTable(title='Smart Management')
+    # element attributes changed from id to class in upstream-fine+, capture both with locator
+    flash = FlashMessages('.//div[@id="flash_msg_div"]'
+                          '/div[@id="flash_text_div" or contains(@class, "flash_text_div")]')
+
+
+class ClusterView(BaseLoggedInPage):
+    """Base view for all the cluster views"""
+    @property
+    def in_cluster(self):
+        """Determine if the browser has navigated to the Cluster page"""
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Compute', 'Infrastructure', 'Clusters'] and
+            # TODO: needs to be converted to Widgetastic once we have a replacement
+            match_location(controller='ems_cluster', title='Clusters'))
+
+
+class ClusterAllView(ClusterView):
+    """The all view page for clusters"""
+    @property
+    def is_displayed(self):
+        """Determine if this page is currently being displayed"""
+        return (
+            self.in_cluster and
+            self.entities.title.text == 'Clusters')
+
+    toolbar = View.nested(ClusterToolbar)
+    entities = View.nested(ClusterEntities)
+    paginator = View.nested(PaginationPane)
+
+
+class ClusterDetailsView(ClusterView):
+    """The details page of a cluster"""
+    @property
+    def is_displayed(self):
+        """Determine if this page is currently being displayed"""
+        expected_title = '{} (Summary)'.format(self.context['object'].name)
+        return (
+            self.in_cluster and
+            self.entities.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
 
+    toolbar = View.nested(ClusterDetailsToolbar)
+    sidebar = View.nested(ClusterDetailsAccordion)
+    entities = View.nested(ClusterDetailsEntities)
 
-class InfraClusterTimelinesView(TimelinesView, BaseLoggedInPage):
+
+class ClusterTimelinesView(TimelinesView, ClusterView):
+    """The timelines page of a cluster"""
     @property
     def is_displayed(self):
-        return self.logged_in_as_current_user and \
-            self.navigation.currently_selected == ['Compute', 'Infrastructure', 'Clusters'] and \
-            super(TimelinesView, self).is_displayed
+        """Determine if this page is currently being displayed"""
+        return (
+            self.in_cluster and
+            super(TimelinesView, self).is_displayed)
 
 
 class Cluster(Pretty, Navigatable):
     """ Model of an infrastructure cluster in cfme
 
-    Args:
-        name: Name of the cluster.
-        provider: provider this cluster is attached to.
+    :param name: Name of the cluster.
+    :param provider: provider this cluster is attached to.
 
-    Note:
+    .. _note:
```

Comment:
Whilst there may be preference here, please stick to the current way of expressing docstrings to maintain consistency